### DISCUSSION TITLE
zksync-crypto in next.js cant build.  node.js and web.js should merge.

### DISCUSSION THREAD

**Discussion by Alan-Cn** [zkSync-Community-Hub/zksync-developers#286] [SDKs] [answered]
### Environment

Mainnet

### Acknowledgement

- [ ] I understand that issues are expected with the Go SDK as they are being updated.

### Issue Description

In next.js, it is impossible to accurately obtain whether the crypto file type is node or web, resulting in packaging failure.

### Expected Behavior

The packaging should be successful. Is it possible not to distinguish the file type, but to distinguish the node or web environment in the file?

### Code Example

'use client'

import zk from 'zksync-crypto' or import zk from 'zksync-crypto/dist/zksync-crypto-web'

export default function Bridge() {
  console.log(zk)
  return null
}

and  => npm run build

will find a error: 
    1. Error: Cannot find module 'util'
    2. ReferenceError: window is not defined

**Comment by uF4No** [comment]
Hi @Alan-Cn What are you trying to build? `zksync-crypto` is a quite old package and you might want to use `zksync-ethers` instead.

I'll leave you a link to our docs in case it helps you: 

- zksync-ethers v5: https://docs.zksync.io/build/sdks/js/getting-started.html
- zksync-ethers v6: https://docs.zksync.io/build/sdks/js/zksync-ethers/getting-started.html

**Reply by Alan-Cn** [reply]
What I need is functional support for zksync-lite. Can the two copies you sent support zksync-lite?

**Reply by uF4No** [reply, marked as answer]
No, `zksync-ethers` is for zkSync Era.

For zkSync Lite, have you tried following the examples in [this section of the docs](https://docs.lite.zksync.io/api/sdk/js/tutorial/)? Also, [check out this appendix](https://docs.lite.zksync.io/api/sdk/js/browser-bundled/) to run the zksync lite library in the browser.

**Comment by discpenromo** [comment]
To resolve the packaging issue, you can conditionally import the module based on the environment. For example: let zk;

if (typeof window !== 'undefined') {
  // Client-side code
  zk = require('zksync-crypto/dist/zksync-crypto-web');} else {
  // Server-side code
  zk = require('zksync-crypto');


export default function Bridge() {
  console.log(zk);
  return null;

This way, the appropriate module will be imported based on whether the code is running in a Node.js or browser environment.

**Comment by sayloonkenneth** [comment]
To resolve packaging issues in Next.js with zksync-crypto, consider conditional imports based on the environment.
 You can use dynamic imports or environment variables to determine whether to import 'zksync-crypto' or 'zksync-crypto/dist/zksync-crypto-web'.
 This approach allows you to handle both node and web environments without explicitly distinguishing the file type.

**Comment by cbaca9** [comment]
To resolve the issue, you can conditionally import modules based on the environment using techniques such as dynamic imports or environment variables. This allows you to distinguish between Node.js and web environments and import the appropriate module accordingly.

**Comment by segheamulchoidiaz** [comment]
It seems that the issue is related to the inability to accurately determine the crypto file type in a Next.js project, resulting in packaging failure. The expected behavior is for the packaging to be successful. One possible solution could be to distinguish the node or web environment in the file instead of trying to distinguish the file type. The provided code example shows the use of zksync-crypto and the error message indicates that the 'util' module cannot be found and that window is not defined. Without further information or context, it is difficult to provide a more detailed answer.
